### ISSUE TITLE
Support DELETE and INSERT operations in Query-to-File

### ISSUE THREAD

**Issue by vogti** [polypheny/Polypheny-DB#298]
Query-to-File does not support deleting a database record (deleting the file) and inserting new rows (by adding a folder containing files).

**Comment by absh2702** [comment]
Hi @vogti,
I would like to work on this issue. I have already built Polypheny-DB and Query-to-File projects in IntelliJ and am able to successfully run them. I just want to ask if I switch to a Linux machine to work on the Query-to-File project as this project currently does not support Windows OS (Issue #297) ? Any other documents/links to refer to while fixing this issue would be much appreciated.
Thank you.

**Comment by vogti** [comment]
Hi @absh2702, thank you for taking care of this. I have assigned the issue to you.
Have you already tried the Windows Subsystem for Linux? This might work as well. There isn't really any else you need to read, but you can always ask if you have questions.

**Comment by absh2702** [comment]
Hi @vogti,
Thank you for suggesting WSL. I am facing an issue that IntelliJ FREEZES while opening projects via WSL. I am not able to find any solution for this. What do you suggest in this regard?
Thank you.

**Comment by vogti** [comment]
Have you followed this guide? https://www.jetbrains.com/help/idea/how-to-use-wsl-development-environment-in-product.html

**Comment by absh2702** [comment]
Yes, I followed the same guide. After this, I was able to successfully build and run the projects. But, sometimes the IntelliJ gets frozen and then I have to forcefully terminate IntelliJ. This issue mainly arrives if I try to open the second project. Like, if I open Query-to-File after running the instance of the Polypheny-DB, the IDE just freezes.

**Comment by absh2702** [comment]
Hi @vogti, Will it be possible for you to share a video or a documentation on how to run an update query on Query-to-File Interface. I went through the `ResultFS` and `QueryInterface` classes and understood the required methods but I would like to debug and know, how the query is executing. This will help me understand the workflow of the Query-to-File interface in a better way. Thanks.

**Comment by aniket2405** [comment]
Hello, I would like to contribute to this issue, if it is still open. Thanks!

**Comment by vogti** [comment]
Hi @aniket2405, welcome to Polypheny! It would be great if you can take care of this issue. I have just assigned it to you. Importing [Polypheny Query-to-File](https://github.com/polypheny/Query-to-File) in your favorite IDE is similar as it is described for Polypheny-DB [here](https://github.com/polypheny/Polypheny-DB/discussions/286). The main-method is located in `src/main/java/org/polypheny/qtf/Main.java`. Please be aware that you need to have a running instance of Polypheny-DB and follow the instructions in the [readme](https://github.com/polypheny/Query-to-File).

Please be aware, that Query-to-File it is currently not supported on Windows (see #297). Query to File is still in beta. There is only very limited documentation. Therefore, please ask here or open a thread in the [discussions forum](https://github.com/polypheny/Polypheny-DB/discussions) if you have any questions or experience any issues.

**Comment by AniketNS** [comment]
Hello @vogti, I would like to contribute to this issue, if it is still open. Please let me know.

**Comment by riddhishrma** [comment]
Hello, I would like to contribute for this issue, would you like to assign me?

**Comment by Rahulxx01** [comment]
If this issue is not assigned to anyone can I work on this issue as a part of GSOC 2024?

**Comment by Heachy** [comment]
Hello @vogti, 

I can now run Polypheny-DB and Query-to-File projects on Windows. But when I click to open the folder on the Query-to-File page, it shows that there is no such folder. If you change the path to an existing folder, it will not run and the mount failed.

This is to set the return value of `QTFConfig.getMountPoint` to a non-existent folder path. It can run successfully, but when opening the folder, it shows that the folder does not exist.
![UCA`Z%1WWZ5%MJCS21ZON)J](https://github.com/polypheny/Polypheny-DB/assets/90972647/633a0d03-312f-4cf9-be3b-7d94a774f337)

This is to set the return value of QTFConfig.getMountPoint to "`D:\\temp", "test" + File.separator + "qtf\\\\`". When clicking to open the file, it shows that the folder cannot be entered.
![image](https://github.com/polypheny/Polypheny-DB/assets/90972647/bbf96e70-d3ca-4590-9f26-df3a70fbc626)

This is to set the return value of `QTFConfig.getMountPoint` to the existing folder path. It cannot run successfully and the mount fails.
![image](https://github.com/polypheny/Polypheny-DB/assets/90972647/097269f0-3089-4a41-94c1-0b794d3c0a8e)

I've tried multiple paths, but none of them seem to work. The `mount` method of viewing the source code is to create a folder that does not exist as a cache. So is the fact that the folder cannot be opened already exists or is it my problem? QAQ

**Comment by vogti** [comment]
Hi @Heachy and welcome to Polypheny! Cool that you are looking into this.
As mentioned on https://github.com/polypheny/Query-to-File under Limitations, Query-to-File does not officially supported on Windows. However, it might work with the Windows subsystem for Linux. But you need to look into this. It would be great if you get it working on Windows and can create an PR.

**Comment by Heachy** [comment]
Thanks for your encouragement! @vogti 

I found that this project is very similar to the official [MemoryFS ](https://github.com/SerCeMan/jnr-fuse/blob/master/src/main/java/ru/serce/jnrfuse/examples/MemoryFS.java)example in using fuse. So I used maven to build the project and run the example, and found that I could open the file created by fuse. But when I use gradle to build and run the same example, it cannot be opened.

gradle：
![image](https://github.com/SerCeMan/jnr-fuse/assets/90972647/abb0d60b-1b73-4e2a-8ec9-392529e216ef)
maven:
![image](https://github.com/SerCeMan/jnr-fuse/assets/90972647/45da87e8-8df0-4efd-83d3-53bac50bd66e)
Because I am not very proficient in using gradle, I thought at first that I did not configure some environments. But when I copy the sample to this project and run it, it still doesn't work. And when I print information in some places, garbled characters appear. What are your thoughts on this?

I will continue to research and see if you have any suggestions, please let me know, thank you!

**Comment by Heachy** [comment]
> Hi @Heachy and welcome to Polypheny! Cool that you are looking into this. As mentioned on https://github.com/polypheny/Query-to-File under Limitations, Query-to-File does not officially supported on Windows. However, it might work with the Windows subsystem for Linux. But you need to look into this. It would be great if you get it working on Windows and can create an PR.

I tried to use [maven ](https://github.com/Heachy/Query-to-File/tree/maven)to rebuild the project, and finally it can run on Windows！
![U SI`6WZKYP M KCSY@V1~6](https://github.com/polypheny/Polypheny-DB/assets/90972647/f93d448d-d8e1-497e-8def-6f4204ca9621)


Stick to using gradle for this project or change to maven?
